src/Utils/Hooks/useImagePerformanceObserver.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useRef } from "react"
+import { sendToVolley, VolleyMetric } from "Server/volley"
+import { getENV } from "Utils/getENV"
+import { GEMINI_CLOUDFRONT_URL } from "Utils/resizer"
+
+const DEVICE_TYPE = getENV("IS_MOBILE") ? "mobile" : "desktop"
```

Comment:
The value of this `getENV` call is request-specific (on the client), is it chill to have this line here (as a constant) vs. _in_ a hook somewhere?

Reply:
Good q: when this is executed on the client, it will just get the value out of Sharify. It gets set here: https://github.com/artsy/force/blob/8c55b9013ae19c889dc66e4011219b90f99a13c5/src/Server/middleware/bootstrapSharifyAndContextLocalsMiddleware.ts#L35-L45

Theoretically this is more useful than some kind of client-side feature detection. Though I wonder if it's possible to detect if you're on wifi or cell or if you have bandwidth saving mode on or something.

Reply:
Regardless; it doesn't need to be in the hook because per-request is fine.

Reply:
Cool, so provided the same key is available server and client-side thru sharify, you can use `getENV` as a constant or wherever in the code without having to worry.